streaming/src/channel/channel.cc

```diff
@@ -0,0 +1,343 @@
+#include "channel.h"
+
+#include <unordered_map>
+namespace ray {
+namespace streaming {
+
+ProducerChannel::ProducerChannel(std::shared_ptr<Config> &transfer_config,
+                                 ProducerChannelInfo &p_channel_info)
+    : transfer_config_(transfer_config), channel_info_(p_channel_info) {}
+
+ConsumerChannel::ConsumerChannel(std::shared_ptr<Config> &transfer_config,
+                                 ConsumerChannelInfo &c_channel_info)
+    : transfer_config_(transfer_config), channel_info_(c_channel_info) {}
+
+StreamingQueueProducer::StreamingQueueProducer(std::shared_ptr<Config> &transfer_config,
+                                               ProducerChannelInfo &p_channel_info)
+    : ProducerChannel(transfer_config, p_channel_info) {
+  STREAMING_LOG(INFO) << "Producer Init";
+}
+
+StreamingQueueProducer::~StreamingQueueProducer() {
+  STREAMING_LOG(INFO) << "Producer Destory";
+}
+
+StreamingStatus StreamingQueueProducer::CreateTransferChannel() {
+  CreateQueue();
+
+  uint64_t queue_last_seq_id = 0;
+  uint64_t last_message_id_in_queue = 0;
+
+  if (!last_message_id_in_queue) {
```

Comment:
last_message_id_in_queue always will be 0, so the first condition could be true forever.